livre/core/cache/CacheStatistics.h

```diff
@@ -70,11 +70,11 @@ class CacheStatistics : public CacheObjectObserver
     CacheStatistics( const std::string& statisticsName,
                      const size_t queueSize );
 
-    void _onLoaded( const CacheObject& cacheObject ) final;
-    void _onUnload( const CacheObject& cacheObject ) final;
+    void _notifyLoaded( const CacheObject& cacheObject );
+    void _notifyUnloaded( const CacheObject& cacheObject );
 
-    void _onCacheMiss( const CacheObject& ) final { ++_cacheMiss; }
-    void _onCacheHit( const CacheObject& ) final { ++_cacheHit; }
+    void _notifyMiss() { ++_cacheMiss; }
+    void _notifyHit() { ++_cacheHit; }
```

Comment:
I would go with a public API and no friends. Obviously statistics are driven by the cache. Users should just get a const ref, disabling the notify methods.